src/mender-update/cli/cli_test.cpp

```diff
@@ -463,6 +475,92 @@ artifact_name=test
 )"));
 }
 
+TEST(CliTest, InstallAndCommitArtifactCheckProvidesDepends) {
+	/* Install two Artifacts. One to install some provides, and the second one to
+	 verify the depends
+	 */
+	mtesting::TemporaryDirectory tmpdir;
+	string artifact = path::Join(tmpdir.Path(), "artifact.mender");
+	ASSERT_TRUE(PrepareSimpleArtifact(tmpdir.Path(), artifact, "test", "", false));
+
+	string update_module = path::Join(tmpdir.Path(), "rootfs-image");
+
+	ASSERT_TRUE(PrepareUpdateModule(update_module, R"(#!/bin/bash
+
+TEST_DIR=")" + tmpdir.Path() + R"("
+
+echo "$1" >> $TEST_DIR/call.log
+
+exit 0
+)"));
+
+	{
+		vector<string> args {
+			"--data",
+			tmpdir.Path(),
+			"install",
+			artifact,
+		};
+
+		mtesting::RedirectStreamOutputs output;
+		int exit_status = cli::Main(
+			args, [&tmpdir](context::MenderContext &ctx) { SetTestDir(tmpdir.Path(), ctx); });
+		EXPECT_EQ(exit_status, 0) << exit_status;
+
+		EXPECT_EQ(output.GetCout(), R"(Installing artifact...
+Update Module doesn't support rollback. Committing immediately.
+Installed and committed.
+)");
+		EXPECT_EQ(output.GetCerr(), "");
+	}
+
+	EXPECT_TRUE(mtesting::FileContainsExactly(
+		path::Join(tmpdir.Path(), "call.log"), R"(ProvidePayloadFileSizes
+Download
+ArtifactInstall
+NeedsArtifactReboot
+SupportsRollback
+ArtifactCommit
+Cleanup
+)"));
+
+	EXPECT_TRUE(VerifyProvides(tmpdir.Path(), R"(rootfs-image.version=test
+rootfs-image.checksum=f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
+artifact_name=test
+)"));
+
+	/* Depend on the wrong checksum */
+	ASSERT_TRUE(PrepareSimpleArtifact(
+		tmpdir.Path(),
+		artifact,
+		"test",
+		"",
+		false,
+		{
+			"rootfs-image.checksum:notyourcorrectchecksum",
+			"artifact_name:test",
+		}));
+
+	{
+		vector<string> args {
+			"--data",
+			tmpdir.Path(),
+			"install",
+			artifact,
+		};
+
+		mtesting::RedirectStreamOutputs output;
+		int exit_status = cli::Main(
+			args, [&tmpdir](context::MenderContext &ctx) { SetTestDir(tmpdir.Path(), ctx); });
+		EXPECT_EQ(exit_status, 1) << exit_status;
+
+		EXPECT_EQ(output.GetCout(), R"(Installing artifact...
+Installation failed. System not modified.
+)");
+		EXPECT_EQ(output.GetCerr(), "");
```

Comment:
No logging at all? Isn't that a bit suspicious?

Reply:
Yes. But it is the same for all the tests.

I did a closer look. And the only logging on the path, which is not showing is from the artifact parser. And the reason it is not showing is that the log level is lower than `Info`. So this looks right to me :)